tests/test_utils.py

```diff
@@ -58,6 +59,167 @@ def test_mpp2common_objective_power(_sample_svs):
         )
 
 
+def test_safe_padded_read_non_int_bounds():
+    """Test safe_padded_read with non-integer bounds."""
+    data = np.zeros((16, 16))
+
+    bounds = (1.5, 1, 5, 5)
+    with pytest.raises(ValueError):
+        utils.image.safe_padded_read(data, bounds)
+
+
+def test_safe_padded_read_negative_padding():
+    """Test safe_padded_read with non-integer bounds."""
+    data = np.zeros((16, 16))
+
+    bounds = (1, 1, 5, 5)
```

Comment:
Aren't these integer bounds?

Reply:
I'm not sure what you are asking here. This is a test for a function which only accepts integer bounds. Are you thinking of the sub-pixel read function? This is the safe padded read function which the sub-pixel read function calls internally with the bounding integer coordinates.

Reply:
It looks fine to me. If I remember correctly, these values were not integers that's why I wrote the comment. It looks fine to me now.